wrapper/src/main/java/software/amazon/jdbc/plugin/bluegreen/BlueGreenConnectionPlugin.java

```diff
@@ -168,10 +168,12 @@ public Connection connect(
             isInitialConnection,
             connectFunc,
             this.pluginService);
-
         if (conn == null) {
 
           this.bgStatus = this.pluginService.getStatus(BlueGreenStatus.class, this.bgdId);
+          if (this.bgStatus == null) {
```

Comment:
`this.endTimeNano` won't be set up in such case.

Reply:
Then the measured time includes an actual connect time. Original idea was to measure that routing time. If it's easy to fix I'd prefer just routing time.